dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java

```diff
@@ -3958,8 +3956,11 @@ public Node visitUnary(UnaryTree tree, Void p) {
                         result = new LocalVariableNode(resultExpr);
                         result.setInSource(false);
                         extendWithNode(result);
+                        createIncrementOrDecrementAssign(tree, expr, isIncrement, isPostfix);
                     } else {
-                        result = unaryAssign;
+                        result =
+                                createIncrementOrDecrementAssign(
+                                        tree, expr, isIncrement, isPostfix);
```

Comment:
```suggestion
                    AssignmentNode unaryAssign =
                            createIncrementOrDecrementAssign(tree, expr, isIncrement, isPostfix);
                    if (!isPostfix) {
                        result = unaryAssign;
                    }
```

Reply:
Would something like this avoid the duplication of the `create...` call?
Maybe a comment could explain why the `unaryAssign` doesn't need to be used.